crates/templates/src/renderer.rs

```diff
@@ -99,3 +112,31 @@ fn string_from_bytes(bytes: &[u8]) -> Option<String> {
         Err(_) => None, // TODO: try other encodings!
     }
 }
+
+enum TemplateParseFailure {
+    UnknownFilter(String),
+    Other(liquid::Error),
+}
+
+lazy_static! {
+    static ref UNKNOWN_FILTER: regex::Regex =
+        regex::Regex::new("requested filter=(\\S+)").expect("Invalid unknown filter regex");
+}
+
+fn understand_liquid_error(e: liquid::Error) -> TemplateParseFailure {
+    let err_str = e.to_string();
+    let err_lines = err_str.lines().collect_vec();
+
+    // They should use typed errors like we, er, don't
+    match err_lines.first() {
```

Comment:
Nit: no need to collect the lines into a `Vec` if you only care about the first one.
```suggestion
    match err_str.lines().next() {
```

Reply:
ooh thanks!